databricks/koalas/series.py

```diff
@@ -4335,6 +4335,63 @@ def pct_change(self, periods=1):
 
         return self._with_new_scol((scol - prev_row) / prev_row)
 
+    def dot(self, other):
+        """
+        Compute the dot product between the Series and the columns of other.
+
+        This method computes the dot product between the Series and another
+        one, or the Series and each columns of a DataFrame.
+
+        It can also be called using `self @ other` in Python >= 3.5.
+
+        Parameters
+        ----------
+        other : Series, DataFrame.
+            The other object to compute the dot product with its columns.
+
+        Returns
+        -------
+        scalar, Series
+            Return the dot product of the Series and other if other is a
+            Series, the Series of the dot product of Series and each rows of
+            other if other is a DataFrame.
+
+        Notes
+        -----
+        The Series and other has to share the same index if other is a Series
+        or a DataFrame.
+
+        Examples
+        --------
+        >>> s = ks.Series([0, 1, 2, 3])
+
+        >>> s.dot(s)
+        14
+
+        >>> s @ s
+        14
+        """
+        if len(self.index) != len(other.index):
+            raise ValueError("matrices are not aligned")
```

Comment:
Hm, does this only check the length? I think it also checks if the values in index are same or not. Also, we don't have to perform count if both are same Series.

Reply:
I agree that it is better to check if both series is identically same,

but don't think we have to compare the values of indexes.

If the length is same, we can just compute the dot product regardless of their values.

Reply:
That's right, thanks for the comment!